File: app/src/main/java/xyz/farshad/vocab/component/DataAdapter/LevelListAdapter.java
package xyz.farshad.vocab.component.DataAdapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.List;

import xyz.farshad.vocab.R;
import xyz.farshad.vocab.activity.WordListActivity;
import xyz.farshad.vocab.model.Level;

/**
 * Created by farshad on 7/15/16.
 */
public class LevelListAdapter extends ArrayAdapter<Level> {

    public LevelListAdapter(Context context, int textViewResourceId, List<Level> objects) {
        super(context, textViewResourceId, objects);
    }

    @Override
    public View getView(final int position, View convertView, ViewGroup parent) {
        View itemView = convertView;
        if (itemView == null) {
            itemView = LayoutInflater.from(getContext()).inflate(R.layout.level_list_view, parent, false);
        }

        Level currentLevel = getItem(position);

        TextView levelText = itemView.findViewById(R.id.level_name_txt);
        levelText.setText(currentLevel.getName());

        // item click listener
        itemView.setOnClickListener(view -> {
            Level clickedLevel = getItem(position);

            //start word list activity
            Intent wordListIntent = new Intent(getContext(), WordListActivity.class);
            wordListIntent.putExtra("levelId", clickedLevel.getId());
            wordListIntent.putExtra("levelName", clickedLevel.getName());
            getContext().startActivity(wordListIntent);
        });

        return itemView;
    }
}

File: app/src/main/java/xyz/farshad/vocab/component/DataAdapter/LevelListAdapter.kt
package xyz.farshad.vocab.component.DataAdapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import xyz.farshad.vocab.R
import xyz.farshad.vocab.activity.WordListActivity
import xyz.farshad.vocab.model.Level

/**
 * Created by farshad on 7/15/16.
 */
class LevelListAdapter(context: Context, textViewResourceId: Int, objects: List<Level>) : ArrayAdapter<Level>(context, textViewResourceId, objects) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var itemView = convertView
        if (itemView == null) {
            itemView = LayoutInflater.from(context).inflate(R.layout.level_list_view, parent, false)
        }

        val currentLevel = getItem(position)

        val levelText = itemView!!.findViewById<TextView>(R.id.level_name_txt)
        levelText.text = currentLevel!!.name

        // item click listener
        itemView.setOnClickListener { view ->
            val clickedLevel = getItem(position)

            //start word list activity
            val wordListIntent = Intent(context, WordListActivity::class.java)
            wordListIntent.putExtra("levelId", clickedLevel!!.id)
            wordListIntent.putExtra("levelName", clickedLevel.name)
            context.startActivity(wordListIntent)
        }

        return itemView
    }
}
